src/gt4py/next/iterator/type_inference.py

```diff
@@ -417,7 +417,25 @@ def visit(self, node, **kwargs) -> typing.Any:
         return result
 
     def visit_Sym(self, node: ir.Sym, **kwargs) -> Type:
-        return TypeVar.fresh()
+        result = TypeVar.fresh()
+        if node.kind:
+            kind = {"Iterator": Iterator(), "Value": Value()}[node.kind]
```

Comment:
It's not very important, but I think it would better to use explicit pattern matching:
```suggestion
        match node.kind:
            case "Iterator":
                kind = Iterator()
            case "Value":
                kind = Value()
            case _:
                raise AssertionError(...)
```

Reply:
I see the point, but I like the less verbose version more.

Reply:
Honestly, for just two cases, I don't think the dict version brings any value. Even a simple `if-else` would be better.